### PR TITLE
chore: Examples Pruning

### PR DESCRIPTION
**Pull request by MikhailKardash**
## Description

replaces #7918 because Github is being unruly

Move pruning-marked examples to a legacy directory. Remove their download links and references from docs. Remove their tests from CI. 

Remove tutorials and model hub v1 examples docs.

## Test Plan

CI passes

## Commentary (optional)

Legacy folder should be migrated to a new repository in a follow-up PR.

Maybe think of a new way to organize examples.

## Checklist

- [x] Changes have been manually QA'd
- [x] Docs and github references to the legacy examples have been resolved
- [ ] Blog post references to the legacy examples have been resolved
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket
MLG-869

### CODE REVIEW

File: docs/tutorials/index.rst [hunk 1]
```diff
@@ -20,16 +20,8 @@ you'll learn how to perform the following tasks:
...
-| Title                           | Description                                                  |
-+=================================+==============================================================+
-| :doc:`pytorch-mnist-tutorial`   | Based on the `PyTorch MNIST example`_, this tutorial shows   |
-|                                 | you how to port a simple image classification model for the  |
-|                                 | MNIST dataset.                                               |
-+---------------------------------+--------------------------------------------------------------+
-| :doc:`tf-mnist-tutorial`        | The TensorFlow Keras Fashion MNIST tutorial describes how to |
-|                                 | port a ``tf.keras`` model to Determined.                     |
-+---------------------------------+--------------------------------------------------------------+
+To get started with a :ref:`Trial API <high-level-apis>`, visit the :ref:`pytorch-mnist-tutorial`.
```
Comment (dzhu) [inline docs/tutorials/index.rst:23]: In this case, the "the" makes the link text part of the sentence and not just a location, so the text should be written out. (Or just remove the "the".)

```suggestion
To get started with a :ref:`Trial API <high-level-apis>`, visit the :ref:`PyTorch MNIST tutorial <pytorch-mnist-tutorial>`.
```

Comment (tara-det-ai) [reply docs/tutorials/index.rst:23]: good catch

**Review by tara-det-ai — APPROVED** [review]
LGTM

File: docs/example-solutions/examples.rst [hunk 2]
```diff
@@ -19,7 +19,8 @@ For example, here is how to train the ``mnist_pytorch`` example with a fixed set
    cd mnist_pytorch
    det experiment create const.yaml .
 
-For an introduction to using the training API, please visit the Training APIs section.
+For an introduction to using the training APIs, please visit :ref:`Training APIs
```
[the remark this answers is not in the document]
Comment (tara-det-ai) [reply docs/example-solutions/examples.rst:22]: For an introduction to using the training APIs, visit the :ref:`Training API Guides <apis-howto-overview>`.

**Review by azhou-determined — APPROVED** [review]
